bundle/Command/TagContentByTypesCommand.php

```diff
@@ -0,0 +1,176 @@
+
+            $this->repository->beginTransaction();
+
+            foreach ($searchResults->searchHits as $searchHit) {
+                /** @var \Ibexa\Contracts\Core\Repository\Values\Content\Content $content */
+                $content = $searchHit->valueObject;
+
+                if ($content->getField($fieldIdentifier) === null) {
+                    continue;
+                }
+
+                try {
+                    if (!$content->getField($fieldIdentifier)->value instanceof TagFieldValue) {
+                        $this->style->warning(sprintf('Field with identifier %s must be a type of eztags', $fieldIdentifier));
+
+                        continue;
+                    }
+
+                    $alreadyAssignedTags = $content->getFieldValue($fieldIdentifier)->tags;
+                    $tag = $this->tagsService->loadTag($this->getTagId((int) $input->getOption('tag-id')));
+                    $tagsToAssign = array_filter($alreadyAssignedTags, static fn ($alreadyAssignedTag) => $tag->id !== $alreadyAssignedTag->id);
+                    $tagsToAssign[] = $tag;
+
+                    $this->repository->sudo(
+                        function () use ($content, $fieldIdentifier, $tagsToAssign): void {
+                            $contentDraft = $this->contentService->createContentDraft($content->contentInfo);
+                            $contentUpdateStruct = $this->contentService->newContentUpdateStruct();
+
+                            $contentUpdateStruct->setField($fieldIdentifier, new TagFieldValue($tagsToAssign));
+
+                            $this->contentService->updateContent($contentDraft->versionInfo, $contentUpdateStruct);
+                            $this->contentService->publishVersion($contentDraft->versionInfo);
+                        },
+                    );
+
+                    $this->style->progressAdvance();
+                } catch (Throwable $t) {
+                    $this->style->error($t->getMessage());
+
+                    $this->repository->rollback();
+
+                    continue;
+                }
+            }
+            $this->repository->commit();
+        }
+
+        $this->style->progressFinish();
+        $this->style->success('Tags assigned successfully');
+
+        return Command::SUCCESS;
+    }
+
+    private function getParentLocation(int $parentLocationInput): int
+    {
+        if ($parentLocationInput < 1) {
+            throw new InvalidOptionException(
+                sprintf("Argument --parent-location must be an integer > 0, you provided '%s'", $parentLocationInput),
+            );
+        }
+
+        return $parentLocationInput;
+    }
+
+    /**
+     * @return array<string>
+     */
+    private function getContentTypes(string $contentTypesInput): array
+    {
+        return $this->parseCommaDelimited($contentTypesInput);
+    }
+
+    private function getTagId(int $tagIdInput): int
```

Comment:
All of these method arguments should not have `Input` suffix.

Reply:
solved in fbca72c7